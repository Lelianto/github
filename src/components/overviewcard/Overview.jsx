import React from 'react';
import { stars } from '../../assets/svg';
import colors from '../../assets/colors.json'

const Overview = ({ repository }) => {
	const setStyle = (language) => {
		if (colors[`${language}`]) {
			return { backgroundColor: colors[`${language}`].color }
		}
	}

	const language = (languageExist) => {
		if (languageExist) {
			return (
				<>
					<div className="mr-1"><span className="inline-block h-3 w-3 rounded-full" style={setStyle(languageExist)}></span></div>
					<div id="language" className="mr-3">
						{languageExist}
					</div>
				</>
			)
		} else {
			return <></>
		}
	}
	return (
		<>
			<div className="flex w-full flex-col justify-between border rounded-sm p-3 h-full">
				<div className="text-left text-base text-blue-700 font-bold break-words">
					<a
						href={`https://github.com/${repository.full_name}`}
						target="_blank"
						className="text-blue-700 font-bold"
						rel="noreferrer"
					>
						<span id="repoName">{repository.name}</span>
					</a>
				</div>
				<div id="description" className="text-left text-base text-gray-500 break-words">
					{repository.description}
				</div>
				<div className="flex mt-3">
					{language(repository.language)}
					<div className="flex">
						<div className="relative m-auto ml-0 mr-1">
							{stars()}
						</div>
						<div id="count" className="relative m-auto ml-0 mr-1">
							{repository.stargazers_count}
						</div>
					</div>
				</div>
			</div>
		</>
	)
}

export default Overview;