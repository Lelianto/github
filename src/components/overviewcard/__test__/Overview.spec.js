import Overview from "../Overview";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe("Overview Card: Check content", () => {
	let wrapper;
	let mockProps = {
		id: 237374481,
		name: 'alta_final_fe',
		full_name: 'Lelianto/alta_final_fe',
		stargazers_count: "0",
		language: 'JavaScript',
		description: "React testing using jest in Overview Card"
	}

	beforeEach(() => {
		wrapper = shallow(<Overview repository={mockProps} />)
	});

	it('Renders correctly', () => {
		const tree = renderer.create(<Router><Overview repository={mockProps} /></Router>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Check repository name", () => {
		const repoName = wrapper.find("#repoName")
		expect(repoName.text()).toEqual(mockProps.name)
	})

	it("Check repository description", () => {
		const description = wrapper.find("#description")
		expect(description.text()).toEqual(mockProps.description)
	})

	it("Check repository count", () => {
		const count = wrapper.find("#count")
		expect(count.text()).toEqual(mockProps.stargazers_count)
	})

	it("Check repository main language", () => {
		const language = wrapper.find("#language")
		expect(language.text()).toEqual(mockProps.language)
	})
})