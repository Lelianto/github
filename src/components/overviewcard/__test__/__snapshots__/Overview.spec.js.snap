// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Card: Check content Renders correctly 1`] = `
<div
  className="flex w-full flex-col justify-between border rounded-sm p-3 h-full"
>
  <div
    className="text-left text-base text-blue-700 font-bold break-words"
  >
    <a
      className="text-blue-700 font-bold"
      href="https://github.com/Lelianto/alta_final_fe"
      rel="noreferrer"
      target="_blank"
    >
      <span
        id="repoName"
      >
        alta_final_fe
      </span>
    </a>
  </div>
  <div
    className="text-left text-base text-gray-500 break-words"
    id="description"
  >
    React testing using jest in Overview Card
  </div>
  <div
    className="flex mt-3"
  >
    <div
      className="mr-1"
    >
      <span
        className="inline-block h-3 w-3 rounded-full"
        style={
          Object {
            "backgroundColor": "#f1e05a",
          }
        }
      />
    </div>
    <div
      className="mr-3"
      id="language"
    >
      JavaScript
    </div>
    <div
      className="flex"
    >
      <div
        className="relative m-auto ml-0 mr-1"
      >
        <svg
          aria-hidden="true"
          className="octicon octicon-star text-icon-tertiary"
          data-view-component="true"
          height="16"
          version="1.1"
          viewBox="0 0 16 16"
          width="16"
        >
          <path
            d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className="relative m-auto ml-0 mr-1"
        id="count"
      >
        0
      </div>
    </div>
  </div>
</div>
`;
